### ISSUE TITLE
InputDecorator 'errorText' should have 'textAlign' property

### ISSUE THREAD

**Issue by kostaa** [flutter/flutter#11068]
When used with the TextFiled the InputDecorator 'errorText' uses the TextFiled's 'textAlign' property (I suspect).

This is a problem when the TextFiled is used to show numerical values in which case the value should be right-aligned but the error message should be left-aligned.

**Comment by Hixie** [comment]
cc @HansMuller

**Comment by HansMuller** [comment]
This makes sense. We could an errorAlign property or just enable providing an error widget.

**Comment by jonas-zebari** [comment]
Is there an update on this issue? I have also run into the need for explicitly specifying the text alignment for the field input and the error text separately.

**Comment by esDotDev** [comment]
I think the errorWidget approach makes sense. Most of what is useful about the TextInput's built-in error handling is the way the label interacts with the border, and animates it's color and position.

The error messages are simple widgets, we can provide those ourselves and it is much more flexible.

For example, I'm stuck right now cause I need a 2 line error message, which is a dead end with the existing TextInput. Now I basically need to hack the widget to get the error outline to show, while providing my own error message...

**Comment by erf** [comment]
As i use `textAlign: TextAlign.center` for the `TextFormField`, i would like the error text to also be center aligned.

A solution could be to either return a `Widget` in the `validator` or set in `errorText` ( instead of a string ), or add a `errorAlignment` property.

**Comment by wilburt** [comment]
Please, is there any update on this?

**Comment by LockedThread** [comment]
I'm with @wilburt on this. When will this be added? This is pretty necessary and there are multiple issues open related to this.

**Comment by oaajunior** [comment]
Please, is there any new sfor this change?

**Comment by Katekko** [comment]
Any news to put that?

**Comment by victorkarangwa4** [comment]
Any update about this issue?

**Comment by SAGARSURI** [comment]
Any update on this?

**Comment by ronny-sugianto** [comment]
any update ?

**Comment by killerm250m** [comment]
any update ?

**Comment by ronny-sugianto** [comment]
I have an easy solution & I hope this helps, we just need to create a widget with an example like this


    @override
    Widget build(BuildContext context) {
      WidgetsBinding.instance.addPostFrameCallback((_) {
        setState(() {
          _isErrorWithValidatorKey = _fieldKey?.currentState?.hasError ?? false;
          _readOnly = _formState?.readOnly == true || widget.isReadOnly;
          _hasError = _isErrorWithValidatorKey;
        });
      });
      
      return Column(
         mainAxisSize: MainAxisSize.min,
         children: [
            Container(
              margin: const EdgeInsets.only(bottom: 2),
              child: TextFormField(
              decoration: InputDecoration(
                      errorStyle: TextStyle(
                        height: 0,
                        fontSize: 0,
                      ),
                   ),
                ),
              ),
            if (_hasError) ...[
                Padding(
                  padding: const EdgeInsets.symmetric(vertical: 6),
                  child: Align(
                    alignment: widget.errorMessageAlign,
                    child: Text(
                      "${_fieldKey.currentState.errorText}",
                      style: TextStyle(
                        color: Palette.red,
                        fontWeight: FontWeight.bold,
                        fontSize: 10,
                      ),
                    ),
                  ),
                )
            ],
        ],
      );
    }

**Comment by koldoon** [comment]
Turn errorText into errorWidget would be perfect.

**Comment by jbryanh** [comment]
Looks like progress!  At minimum it seems to make sense to have it follow the form field alignment.

**Comment by tapizquent** [comment]
Any updates on this?

**Comment by wemgl** [comment]
Any updates on this?

**Comment by ScottishRoss** [comment]
Has this seriously not been addressed for nearly 4 years?

**Comment by IT19951836-Lakshan** [comment]
only one way to do that  put spaces before typing the text then you can align to rigth side . i think no other ways to do that 
![image](https://user-images.githubusercontent.com/87911514/151377766-8c430578-7361-47f8-bffb-263cbc22c36b.png)
![image](https://user-images.githubusercontent.com/87911514/151377788-f3342497-a99a-456e-a149-b49bf7af21bf.png)
I saw lot of people unlike my previous comment but i think there is not exact solution . so if you want to ailing center or rigth side to your text you have to do what i said

**Comment by tapizquent** [comment]
> only one way to do that  put spaces before typing the text then you can align to rigth side . i think no other ways to do that 
> 
> ![image](https://user-images.githubusercontent.com/87911514/151377766-8c430578-7361-47f8-bffb-263cbc22c36b.png)
> 
> ![image](https://user-images.githubusercontent.com/87911514/151377788-f3342497-a99a-456e-a149-b49bf7af21bf.png)
> 
> I saw lot of people unlike my previous comment but i think there is not exact solution . so if you want to ailing center or rigth side to your text you have to do what i said
> 
> 

This is not a viable solution. If you're working with different screen sizes or platforms, it will only be (kinda) centered on one device and completely misaligned on other devices

**Comment by IT19951836-Lakshan** [comment]
> > only one way to do that  put spaces before typing the text then you can align to rigth side . i think no other ways to do that
> > ![image](https://user-images.githubusercontent.com/87911514/151377766-8c430578-7361-47f8-bffb-263cbc22c36b.png)
> > ![image](https://user-images.githubusercontent.com/87911514/151377788-f3342497-a99a-456e-a149-b49bf7af21bf.png)
> > I saw lot of people unlike my previous comment but i think there is not exact solution . so if you want to ailing center or rigth side to your text you have to do what i said
> 
> This is not a viable solution. If you're working with different screen sizes or platforms, it will only be (kinda) centered on one device and completely misaligned on other devices

you're right  my friend but there is no exact way to do that, I think it might be help to demonstration purpose app creator like university student.(if you can do somthing, better than do nothing) as you said it's not good for industrial level. its a tempory solution.

**Comment by misici234** [comment]
I run Flutter 2.8.1 and this issue has not been resolved after more than 4.5 years. Please, what do we have to do so that you fix it?

**Comment by esDotDev** [comment]
Probably someone will need to land a PR if they want this fixed.

**Comment by sjmamani** [comment]
For my case, I got a workaround solution using `Directionality` widget

```
const Directionality(
  textDirection: TextDirection.rtl, // align errorText to the right
  child: TextField(
    textDirection: TextDirection.ltr, // keep the input text aligned to the left
    decoration: InputDecoration(errorText: 'Field required'),
  ),
),
```

<img width="299" alt="image" src="https://user-images.githubusercontent.com/42153383/159783612-d9f7d4a0-2376-4f3f-8b86-370b74040862.png">

**Comment by whygee-dev** [comment]
any updates?

**Comment by Michele-x98** [comment]
I got the solution, i'm trying to make a PR but i don't have enough time to make the tests. 
The solution consist in a WidgetBuilder function inside the InputDecoration class that take in input the errorText String when not Null and we can build a widget from it. If someone want to help me on how to tests it i really appreciate it.

**Comment by CordMemescape** [comment]
> For my case, I got a workaround solution using `Directionality` widget
> 
> ```
> const Directionality(
>   textDirection: TextDirection.rtl, // align errorText to the right
>   child: TextField(
>     textDirection: TextDirection.ltr, // keep the input text aligned to the left
>     decoration: InputDecoration(errorText: 'Field required'),
>   ),
> ),
> ```
> 
> <img alt="image" width="299" src="https://user-images.githubusercontent.com/42153383/159783612-d9f7d4a0-2376-4f3f-8b86-370b74040862.png">
> 
> **NOTE:** `Directionality` widget allows to align only to the left or to the right.

Nice idea but it also aligns the InputDecoration's labelText to the right and there doesn't appear to be a way to align that back to the left. So, a good workaround if you don't use labelText. Not so good if you do.

The right solution would be to simply replace the errorText string with an errorText widget and allow the user to decide.

**Comment by HansMuller** [comment]
There's a PR underway that's attempting to provide support for specifying an error widget. Looks like it could use some help: https://github.com/flutter/flutter/pull/104165

**Comment by Julitech** [comment]
Any updates on this??

**Comment by HansMuller** [comment]
A fix was started in January and seemed to be getting pretty close: https://github.com/flutter/flutter/pull/118610. Unfortunately the dev hasn't been active since early March.

CC @Renzo-Olivares, @justinmc, @KishanBusa8

**Comment by justinmc** [comment]
If @kishanbusa8 doesn't respond, then someone should probably take over the PR.  I think it was pretty close, from my memory.

**Comment by bleroux** [comment]
FYI, I filed https://github.com/flutter/flutter/pull/129275 which can be used by @KishanBusa8 to update https://github.com/flutter/flutter/pull/118610 or can become a non WIP PR if @KishanBusa8 does not respond or can not work on the update.

**Comment by justinmc** [comment]
Great, thanks!  Let's give it until next week maybe and then move forward with your PR if we haven't heard back.
